Source/Charts/Charts/ChartViewBase.swift

```diff
@@ -855,47 +855,59 @@ open class ChartViewBase: NSUIView, ChartDataProvider, AnimatorDelegate
     
     public enum ImageFormat
     {
-        case jpeg
+        /// - parameter quality: compression quality for lossless formats (JPEG)
+        case jpeg(quality: Double)
         case png
     }
     
-    /// Saves the current chart state with the given name to the given path on
-    /// the sdcard leaving the path empty "" will put the saved file directly on
-    /// the SD card chart is saved as a PNG image, example:
-    /// saveToPath("myfilename", "foldername1/foldername2")
+    /// Saves the current chart state with the given name to the given path, examples:
+    /// save(to: "./out/filename.jpg", format: .jpeg(quality: 90.0))
+    /// save(to: "./out/filename.png", format: .png)
     ///
     /// - parameter to: path to the image to save
     /// - parameter format: the format to save
-    /// - parameter compressionQuality: compression quality for lossless formats (JPEG)
     ///
     /// - returns: `true` if the image was saved successfully
-    open func save(to path: String, format: ImageFormat, compressionQuality: Double) -> Bool
+    open func save(to path: String, format: ImageFormat) -> Bool
     {
-		guard let image = getChartImage(transparent: format != .jpeg)
-            else { return false }
+        let saveUrl = URL(fileURLWithPath: path)
+        if #available(iOS 9.0, *)
+        {
+            if saveUrl.hasDirectoryPath
+            {
+                // the path represents a directory. not include filename.
+                return false
+            }
+        }
         
-        var imageData: Data!
+        var imageData: Data?
         switch (format)
         {
         case .png:
+            guard let image = getChartImage(transparent: true) else { return false }
             imageData = NSUIImagePNGRepresentation(image)
-            break
             
-        case .jpeg:
+        case .jpeg(let compressionQuality):
+            guard let image = getChartImage(transparent: false) else { return false }
             imageData = NSUIImageJPEGRepresentation(image, CGFloat(compressionQuality))
-            break
         }
         
-        do
+        if let imageData = imageData
```

Comment:
Swift's try ~ catch dose not handle forced unwrapping error.
on Playground, macOS 10.12.4, Xcode 8.3.2, Swift 3.1
<img width="650" alt="2017-05-24 15 52 19" src="https://cloud.githubusercontent.com/assets/19257572/26390393/59522b8e-4099-11e7-820b-c1fe2472ac53.png">